examples/ble_hid_periph.py

```diff
@@ -0,0 +1,153 @@
+    b'\x09\x38'        #     Usage (Wheel)
+    b'\x15\x81'        #     Logical Minimum (-127)
+    b'\x25\x7F'        #     Logical Maximum (127)
+    b'\x75\x08'        #     Report Size (8)
+    b'\x95\x01'        #     Report Count (1)
+    b'\x81\x06'        #     Input (Data,Var,Rel,No Wrap,Linear,Preferred State,No Null Position)
+    b'\xC0'            #   End Collection
+    b'\xC0'            # End Collection
+    b'\x05\x0C'        # Usage Page (Consumer)
+    b'\x09\x01'        # Usage (Consumer Control)
+    b'\xA1\x01'        # Collection (Application)
+    b'\x85\x03'        #   Report ID (3)
+    b'\x75\x10'        #   Report Size (16)
+    b'\x95\x01'        #   Report Count (1)
+    b'\x15\x01'        #   Logical Minimum (1)
+    b'\x26\x8C\x02'    #   Logical Maximum (652)
+    b'\x19\x01'        #   Usage Minimum (Consumer Control)
+    b'\x2A\x8C\x02'    #   Usage Maximum (AC Send)
+    b'\x81\x00'        #   Input (Data,Array,Abs,No Wrap,Linear,Preferred State,No Null Position)
+    b'\xC0'            # End Collection
+    # b'\x05\x01'        # Usage Page (Generic Desktop Ctrls)
+    # b'\x09\x05'        # Usage (Game Pad)
+    # b'\xA1\x01'        # Collection (Application)
+    # b'\x85\x05'        #   Report ID (5)
+    # b'\x05\x09'        #   Usage Page (Button)
+    # b'\x19\x01'        #   Usage Minimum (\x01)
+    # b'\x29\x10'        #   Usage Maximum (\x10)
+    # b'\x15\x00'        #   Logical Minimum (0)
+    # b'\x25\x01'        #   Logical Maximum (1)
+    # b'\x75\x01'        #   Report Size (1)
+    # b'\x95\x10'        #   Report Count (16)
+    # b'\x81\x02'        #   Input (Data,Var,Abs,No Wrap,Linear,Preferred State,No Null Position)
+    # b'\x05\x01'        #   Usage Page (Generic Desktop Ctrls)
+    # b'\x15\x81'        #   Logical Minimum (-127)
+    # b'\x25\x7F'        #   Logical Maximum (127)
+    # b'\x09\x30'        #   Usage (X)
+    # b'\x09\x31'        #   Usage (Y)
+    # b'\x09\x32'        #   Usage (Z)
+    # b'\x09\x35'        #   Usage (Rz)
+    # b'\x75\x08'        #   Report Size (8)
+    # b'\x95\x04'        #   Report Count (4)
+    # b'\x81\x02'        #   Input (Data,Var,Abs,No Wrap,Linear,Preferred State,No Null Position)
+    # b'\xC0'            # End Collection
+)
+#pylint: enable=line-too-long
+
+hid = HIDService(HID_DESCRIPTOR)
+device_info = DeviceInfoService(software_revision=adafruit_ble.__version__,
+                                manufacturer="Adafruit Industries")
+print(device_info.manufacturer)
+advertisement = ProvideServicesAdvertisement(hid)
+advertisement.appearance = 961
+
+ble = adafruit_ble.BLERadio()
+print(to_hex(bytes(advertisement)))
+if not ble.connected:
+    print("advertising")
+    ble.start_advertising(advertisement)
+else:
+    print("already connected")
+    print(ble.connections)
+
+k = Keyboard(hid.devices)
+kl = KeyboardLayoutUS(k)
+while True:
+    while not ble.connected:
```

Comment:
This spinning on `connected` is becoming a pattern. Maybe a helper `ble.wait_for_connection()` with an optional timeout?